client/my-sites/themes/themes-selection.jsx

```diff
@@ -196,13 +199,20 @@ const ConnectedThemesSelection = connect(
 			number
 		};
 
+		const lastQuery = getLastThemeQuery( state, sourceSiteId ) || query;
```

Comment:
For some reason I was nervous about using `getLastThemeQuery()`, possibly because we added it as a hack for Jetpack search, but the [implementation is straightforward](https://github.com/Automattic/wp-calypso/blob/e9186f5805155e429a7b11d241d76719269a050b/client/state/themes/reducer.js#L340) and it seems to be a good solution here.